charger/openevse.go

```diff
@@ -0,0 +1,446 @@
+func (c *OpenEVSE) Enabled() (bool, error) {
+	overrideResp, err := c.api.GetManualOverrideWithResponse(context.Background())
+
+	if err != nil {
+		return false, err
+	}
+
+	if overrideResp.JSON200 != nil && overrideResp.JSON200.State != nil {
+		switch *overrideResp.JSON200.State {
+		case "disabled":
+			return false, nil
+		case "enabled", "active":
+			return true, nil
+		}
+	}
+
+	statusResp, err := c.api.GetStatusWithResponse(context.Background())
+	if err != nil {
+		return false, err
+	}
+
+	var stateCode int
+	if statusResp.JSON200 != nil && statusResp.JSON200.State != nil {
+		stateCode = *statusResp.JSON200.State
+	} else {
+		stateCode = -1
+	}
+
+	var state bool
+	switch stateCode {
+	case 3, 4:
+		state = true
+	default:
+		configResp, err := c.api.GetConfigWithResponse(context.Background())
+
+		if err != nil {
+			return false, err
+		}
+
+		switch *configResp.JSON200.ChargeMode {
+			case "fast":
+				state = true
+			default:
+				state = false
+		}
+	}
+
+	err = c.SetManualOverride(state)
+
+	return state, err
+}
+
+// Enable implements the api.Charger interface
+func (c *OpenEVSE) Enable(enable bool) error {
+	if enable && c.phases == 0 {
+		_, success, err := c.PerformRAPICommand(c.uri, "$S8 0")
+
+		if err != nil {
+			return err
+		}
+
+		if !success {
+			return fmt.Errorf("failed to turn off three phase auto-switching")
+		}
+	}
+
+	err := c.SetManualOverride(enable)
+
+	return err
+}
+
+// MaxCurrent implements the api.Charger interface
+func (c *OpenEVSE) MaxCurrent(current int64) error {
+	cur := int(current)
+	body := openevse.SetManualOverrideJSONRequestBody{
+		ChargeCurrent: &cur,
+	}
+
+	_, err := c.api.SetManualOverrideWithResponse(context.Background(), body)
+
+	return err
+}
+
+var _ api.Meter = (*OpenEVSE)(nil)
+
+// CurrentPower implements the api.Meter interface
+func (c *OpenEVSE) CurrentPower() (power float64, err error) {
+	resp, err := c.api.GetStatusWithResponse(context.Background())
+	if err != nil {
+		return 0, err
+	}
+
+	phases, err := c.Phases()
+	if err != nil {
+		return 0, err
+	}
+
+	current := float64(phases) * float64(*resp.JSON200.Voltage) * float64(*resp.JSON200.Amp) / 1000
+
+	return current, nil
+}
+
+var _ api.ChargeRater = (*OpenEVSE)(nil)
+
+// ChargedEnergy implements the api.ChargeRater interface
+func (c *OpenEVSE) ChargedEnergy() (float64, error) {
+	resp, err := c.api.GetStatusWithResponse(context.Background())
+	if err != nil {
+		return 0, err
+	}
+
+	return float64(*resp.JSON200.Wattsec) / 3600 / 1000, nil
+}
+
+var _ api.MeterCurrent = (*OpenEVSE)(nil)
+
+// Currents implements the api.MeterCurrent interface
+func (c *OpenEVSE) Currents() (float64, float64, float64, error) {
+	resp, err := c.api.GetStatusWithResponse(context.Background())
+	if err != nil {
+		return 0, 0, 0, err
+	}
+
+	cur := float64(*resp.JSON200.Amp) / 1000
+
+	phases, err := c.Phases()
+	if err != nil {
+		return 0, 0, 0, err
+	}
+
+	if phases == 3 {
+		return cur, cur, cur, nil
```

Comment:
Thats not reliable. 2p car?!

Reply:
The EVSE only has one current transformer for measuring current. 
Itself, internally, if set on 3P, it will just assume the same voltage and current across all 3 phases.

Reply:
Then we cant use the values

Reply:
Got it, thanks.